apps/server/src/modules/tool/context-external-tool/service/tool-reference.service.ts

```diff
@@ -47,4 +56,17 @@ export class ToolReferenceService {
 
 		return toolReference;
 	}
+
+	private async determineToolConfigurationStatusThroughValidation(
+		schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalTool
+	): Promise<ToolConfigurationStatus> {
+		try {
+			await this.schoolExternalToolValidationService.validate(schoolExternalTool);
+			await this.contextExternalToolValidationService.validate(contextExternalTool);
+			return ToolConfigurationStatus.LATEST;
+		} catch (err) {
+			return ToolConfigurationStatus.OUTDATED;
```

Comment:
This is one place but you forgot at least this https://github.com/hpi-schul-cloud/schulcloud-server/blob/e887922a9cadd8198a8830870f79cc51189e282d/apps/server/src/modules/tool/school-external-tool/service/school-external-tool.service.ts#L52-L52

Please use your ide functions and look for other usages of the ToolConfigurationStatus